gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -620,7 +620,7 @@ def post_parse_attrs(self, dataset: xr.Dataset = None) -> dict[str, Any]:
                 # For example "date range" should only be updated after a successful parse
                 update_attrs[attr] = dataset.attrs.pop(attr, None)
 
-        return update_attrs
+        return dataset, update_attrs
```

Comment:
This is still modifying the input dataset during the attrs.pop line. I think if the dataset is copied first using a shallow copy with xarray.Dataset.copy you could run pop without modifying the input.